specutils/io/default_loaders/sdss.py

```diff
@@ -26,29 +27,74 @@
 
 def spec_identify(origin, *args, **kwargs):
     """
-    Check whether given file is FITS and has SDSS-III/IV spec type
+    Check whether given input is FITS and has SDSS-III/IV spec type
     BINTABLE in first extension. This is used for Astropy I/O Registry.
     """
-    with fits.open(args[0]) as hdulist:
-        # Test if fits has extension of type BinTable and check for
-        # spec-specific keys
-        return (hdulist[0].header['TELESCOP'] == 'SDSS 2.5-M' and
-                len(hdulist) > 1 and
-                isinstance(hdulist[1], fits.BinTableHDU) and
-                hdulist[1].header['TTYPE3'] == 'ivar')
+    if isinstance(args[2], fits.hdu.hdulist.HDUList):
+        hdulist = args[2]
+    elif isinstance(args[2], _io.BufferedReader):
+        hdulist = fits.open(args[2])
+    else:
+        hdulist = fits.open(args[0], **kwargs)
+
+    # Test if fits has extension of type BinTable and check for spec-specific keys
+    is_sdss = (hdulist[0].header['TELESCOP'] == 'SDSS 2.5-M' and
+               hdulist[0].header.get('FIBERID', 0) > 0 and
+               len(hdulist) > 1 and
+               isinstance(hdulist[1], fits.BinTableHDU) and
+               hdulist[1].header['TTYPE3'] == 'ivar')
+
+    if not isinstance(args[2], (fits.hdu.hdulist.HDUList, _io.BufferedReader)):
```

Comment:
@weaverba137 is there a reason the hdulist is kept open at all?

Reply:
Just as with the handling in #637 if an open file-like object is passed to the reader (same for the identifiers of course), it is left/returned in its original state.

Reply:
Ah, right. I suppose if it's closed in the identified check, it will not be available to the actually loader function.